jekyll/_cci2/bundling-config.md

```diff
@@ -0,0 +1,905 @@
+
+For instance, the following shows declaring a parameter `foo` in the definition of a command `bar`:
+
+```yaml
+commands:
+  bar:
+    parameters:
+      foo:
+        description: this parameter is used for a thing
+        default: "I'm a default value"
+        type: string
+
+    steps:
+      - run: echo '<< parameters.foo >>'
+```
+
+In the above example a value for the parameter `foo` can be passed when invoking the command. For instance, passing the parameter `foo` would look something like:
+
+```yaml
+jobs:
+  myjob:
+    docker:
+      - image: circleci/ruby:2.4-node
+    steps:
+      - bar:
+          foo: "I'm the passed value"
+```
+
+### Parameter Syntax
+{:.no_toc}
+
+A parameter can have the following keys as immediate children:
+
+| Key Name    | Description                                                                                   | Default value |
+|-------------|-----------------------------------------------------------------------------------------------|---------------|
+| description | Optional. Used to generate documentation for your orb.                                        | N/A           |
+| type        | Required. Currently "string", "boolean", "enum", and "steps" are supported                           | N/A           |
+| default     | The default value for the parameter. If not present, the parameter is implied to be required. | N/A           |
+
+### Parameter Types
+{:.no_toc}
+
+This section describes the types of parameters and their usage.
+
+#### String
+{:.no_toc}
+
+Basic string parameters are described below:
+
+```yaml
+commands:
+  copy-markdown:
+    parameters:
+      destination:
+        description: destination directory
+        type: string
+        default: docs
+    steps:
+      - cp *.md << destination >>
+```
+
+Strings should be quoted if they would otherwise represent another type (such as boolean or number) or if they contain characters that have special meaning in yaml. In all other instances, quotes are optional.
+
+#### Boolean
+{:.no_toc}
+
+Boolean parameters are useful for conditionals:
+
+```yaml
+commands:
+  list-files:
+    parameters:
+      all:
+        description: include all files
+        type: boolean
+        default: false
+    steps:
+      - ls <<# all >> -a <</ all >>
+```
+
+Boolean parameter evaluation is based on the [values specified in YAML 1.1][http://yaml.org/type/bool.html]:
+
+* true: `y` `yes` `true` `on`
+* false: `n` `no` `false` `off`
+
+Capitalized and uppercase versions of the above values are also valid.
+
+#### Steps
+{:.no_toc}
+
+Steps are used when you have a job or command that needs to mix predefined and user-defined steps. When passed in to a command or job invocation, the steps passed as parameters are always defined as an array, even if only one step is provided.
```

Comment:
Is "array" the word we use elsewhere? In YAML-land I see "sequence" used. As long as this is consistent across docs I don't think it matters which we use here, though.

Reply:
I don't think we use either, actually, so we can decide here and now (for now I will leave it as-is)

Reply:
If we're choosing I think I'd pick sequence because that's how the YAML spec talks about it.

Reply:
thanks, changed to sequence in both places